app/controllers/admin/trade_stores_controller.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+module Admin
+  class TradeStoresController < Spree::Admin::ResourceController
+    def index
+      @stores =
+        ::Store.page(params[:page]).per(params[:per_page] || Spree::Config[:orders_per_page])
+    end
+
+    def new
+      @store = ::Store.new
+    end
+
+    def create
+      @store = ::Store.new(store_params)
+      if @store.save
+        redirect_to edit_admin_trade_store_url(@store), notice: t('.created_successfully')
+      else
+        flash.now[:error] = @store.errors.full_messages.join(', ')
+        render :new, status: :unprocessable_entity
+      end
+    end
+
+    def edit
+      @store = ::Store.find(params[:id])
+    end
+
+    def update
+      @store = ::Store.find(params[:id])
+
+      if @store.update(store_params)
+        redirect_to edit_admin_trade_store_url(@store), notice: t('.updated')
+      else
+        flash.now[:error] = @store.errors.full_messages.join(', ')
+        render :edit, status: :unprocessable_entity
+      end
+    end
+
+    private
+
+    def model_class
+      ::Store
```

Comment:
this doesn't look like it's used anywhere

Reply:
well spotted

Reply:
Oh no, seems it is need after all:

```Stores Admin new store creation creates a new Store

     Failure/Error: visit '/admin/trade_stores/new'

     

     NameError:

       uninitialized constant Spree::TradeStore

     # ./spec/features/spree/admin/stores_spec.rb:44:in `block (3 levels) in <top (required)>'
```